src/services/scheduler.ts

```diff
@@ -350,7 +350,11 @@ if (require.main === module) {
     res.sendStatus(200);
   });
 
-  app.listen(port, () => {
-    logger.info(`Scheduler running on port ${port}`);
-  });
+  init()
+    .then(() => {
+      app.listen(port, () => {
+        logger.info(`Scheduler running on port ${port}`);
+      });
+    })
+    .catch(e => respondErrorAndDie(undefined, e));
```

Comment:
We had to revert the change related to this before because it caused unexpected issues in staging/production. Can we move the scheduler changes to a separate PR in case this happens again?